frontend/src/metabase/admin/settings/slack/components/SlackForm/SlackForm.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { t } from "ttag";
+import * as Yup from "yup";
+import Form from "metabase/core/components/Form";
+import FormProvider from "metabase/core/components/FormProvider";
+import FormInput from "metabase/core/components/FormInput";
+import FormSubmitButton from "metabase/core/components/FormSubmitButton";
+import FormErrorMessage from "metabase/core/components/FormErrorMessage";
+import { SlackSettings } from "metabase-types/api";
+import { SlackFormMessage } from "./SlackForm.styled";
+
+const SLACK_SCHEMA = Yup.object({
+  "slack-app-token": Yup.string().required(t`required`),
+  "slack-files-channel": Yup.string().required(t`required`),
+});
+
+export interface SlackFormProps {
+  initialValues: SlackSettings;
+  isReadOnly?: boolean;
+  onSubmit?: (values: SlackSettings) => void;
+}
+
+const SlackForm = ({
+  initialValues,
+  isReadOnly,
+  onSubmit = () => undefined,
+}: SlackFormProps): JSX.Element => {
+  return (
+    <FormProvider
+      initialValues={initialValues}
+      validationSchema={!isReadOnly ? SLACK_SCHEMA : undefined}
+      onSubmit={onSubmit}
```

Comment:
This form doesn't automatically transform `slack-files-channel` to lowercase like the previous form does.
https://github.com/metabase/metabase/pull/26273/files#diff-e9f60e303a2a17b03e990dedd3538da299947464939bb21f779dd54b18c0e2d2L24